boat/src/modules/mutedban.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2018-2021 aetheryx & Cynthia K. Rey
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import type { CommandClient, Guild, Member, MemberPartial } from 'eris'
+import config from '../config.js'
+import { ban } from '../mod.js';
+
+const maxInfractions = 4
+
+async function memberRemove(this: CommandClient, guild: Guild, member: Member | MemberPartial) {
+  if (guild.id !== config.discord.ids.serverId) return
+  if (!('roles' in member)) return
+  if (!member.roles.includes(config.discord.ids.roleMuted)) return
+
+  const bans = (await guild.getBans()).map(ban => ban.user.id)
```

Comment:
Use a promise composition instead, looks less messy than a wrapped await
```suggestion
  const bans = await guild.getBans().then((bans) => bans.map((ban) => ban.user.id))
```